pkg/report-emailer/excel_report_generator.go

```diff
@@ -392,11 +389,38 @@ func GetFilePath(fileName string) string {
 	return filePath
 }
 
+func GetFormattedFileName(schedularName string, schedularDateFormat string, schedularDatePosition string) string {
+	var schedularFileName string
+	//default set schedular file name
+	schedularFileName = schedularName
+	if schedularDateFormat != "" {
+		tCurrentDate := time.Now()
+		if schedularDatePosition == "end" {
+			switch schedularDateFormat {
+			case "MMM YYYY":
+				schedularFileName = schedularName + tCurrentDate.Format("Jan") + " " + tCurrentDate.Format("2006")
```

Comment:
This could be just `schedularFileName = schedularName + tCurrentDate.Format("Jan 2006")`

Does schedularName have a space at the end? Do you want this to look something like `testJan 2006` or `test Jan 2006`?

I would also suggest creating a const to the format (The Go date format strings look kind of weird to non-developers and a seperate const make it clearer in my opinion.

Something like this: https://go.dev/play/p/DhETTN62iCc

Reply:
`Does schedularName have a space at the end? Do you want this to look something like testJan 2006 or test Jan 2006?`

Nice thanks will update now.

`I would also suggest creating a const to the format (The Go date format strings look kind of weird to non-developers and a seperate const make it clearer in my opinion.`

As per @GaryWilletts, we will let user add what they want :
e.g. if the schedule name is “my report*” then the filename will be “my report*01-01-2022” else it will be
"my report01-01-2022"

Reply:
:nit Should be schedul**e**r - sorry, the review popped up and that's all I could see when glancing at the email.

Reply:
Schedular is defined as "of or relating to a [schedule](https://www.merriam-webster.com/dictionary/schedule)"

So I think this a valid usage of the word.

If we wanted to get pedantic about naming I something simpler like this is probably a little clearer?
```
func getFormattedFileName(scheduleName string, dateFormat string, datePosition string) string {
	fileName := scheduleName
	if dateFormat != "" {
		currentDateTime := time.Now()
		if datePosition == "end" {
			fileName = fileName + currentDateTime.Format(dateFormat)
		} else {
			fileName = currentDateTime.Format(dateFormat) + fileName
		}
	}
	return fileName
}
```

Could also do early returns and remove some variables to be slightly more efficient and arguably more readable
```
func getFormattedFileName2(scheduleName string, dateFormat string, datePosition string) string {
	if dateFormat == "" {
		// No date format provided, so use the schedule name as the file name
		return scheduleName
	}

	if datePosition == "end" {
		return scheduleName + time.Now().Format(dateFormat)
	}

	// Default to putting the date first
	return time.Now().Format(dateFormat) + scheduleName
}
```
Overall though I think there's nothing fundamentally and even if the word was genuinely misspelled I don't think there was a risk of anyone miss understanding the code.